packages/berry-pnpify/sources/index.ts

```diff
@@ -0,0 +1,176 @@
+import fs from 'fs';
+import os from 'os';
+import path from 'path';
+
+function readdirNodeModules(pnp: any, dir: string, issuer: string): string[] | undefined {
+  const dirs: { [dir: string]: string[] } = {};
+  const locator = pnp.findPackageLocator(issuer);
+  if (!locator) {
+    return undefined;
+  }
+  const info = pnp.getPackageInformation(locator);
+  if (info.packageLocation + '/' !== issuer) {
+    return undefined;
+  }
+  const packages: string[] = Array.from(info.packageDependencies.keys());
+  packages.forEach((pkg: string) => {
+    dirs[''] = dirs[''] || [];
+    if (pkg[0] === '@') {
+      const [ scopeName, packageName ] = pkg.split('/');
```

Comment:
You should use `structUtils.parseIdent` and test whether the value it returns has a non-null `scope` property (this way the `@` logic is kept in one single place).